integration-tests/src/test/java/io/opentelemetry/OtlpExporterIntegrationTest.java

```diff
@@ -0,0 +1,355 @@
+        OtlpHttpSpanExporter.builder()
+            .setEndpoint(
+                "http://"
+                    + COLLECTOR.getHost()
+                    + ":"
+                    + COLLECTOR.getMappedPort(COLLECTOR_OTLP_HTTP_PORT)
+                    + "/v1/traces")
+            .setCompression("gzip")
+            .build();
+
+    testTraceExport(otlpGrpcTraceExporter);
+  }
+
+  private static void testTraceExport(SpanExporter spanExporter) {
+    SdkTracerProvider tracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
+            .setResource(RESOURCE)
+            .build();
+
+    Span span =
+        tracerProvider
+            .get(OtlpExporterIntegrationTest.class.getName())
+            .spanBuilder("my span name")
+            .addLink(
+                SpanContext.create(
+                    IdGenerator.random().generateTraceId(),
+                    IdGenerator.random().generateSpanId(),
+                    TraceFlags.getDefault(),
+                    TraceState.getDefault()))
+            .startSpan();
+    span.setAttribute("key", "value");
+    span.addEvent("event");
+    span.end();
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(() -> GRPC_SERVER.traceRequests.size() == 1);
+
+    ExportTraceServiceRequest request = GRPC_SERVER.traceRequests.get(0);
+    assertThat(request.getResourceSpansCount()).isEqualTo(1);
+
+    ResourceSpans resourceSpans = request.getResourceSpans(0);
+    assertThat(resourceSpans.getResource().getAttributesList())
+        .contains(
+            KeyValue.newBuilder()
+                .setKey(ResourceAttributes.SERVICE_NAME.getKey())
+                .setValue(AnyValue.newBuilder().setStringValue("integration test").build())
+                .build());
+    assertThat(resourceSpans.getInstrumentationLibrarySpansCount()).isEqualTo(1);
+
+    InstrumentationLibrarySpans ilSpans = resourceSpans.getInstrumentationLibrarySpans(0);
+    assertThat(ilSpans.getInstrumentationLibrary().getName())
+        .isEqualTo(OtlpExporterIntegrationTest.class.getName());
+    assertThat(ilSpans.getSpansCount()).isEqualTo(1);
+
+    io.opentelemetry.proto.trace.v1.Span protoSpan = ilSpans.getSpans(0);
+    assertThat(protoSpan.getName()).isEqualTo("my span name");
+    assertThat(protoSpan.getAttributesList())
+        .isEqualTo(
+            Collections.singletonList(
+                KeyValue.newBuilder()
+                    .setKey("key")
+                    .setValue(AnyValue.newBuilder().setStringValue("value").build())
+                    .build()));
+    assertThat(protoSpan.getEventsCount()).isEqualTo(1);
+    assertThat(protoSpan.getEvents(0).getName()).isEqualTo("event");
+    assertThat(protoSpan.getLinksCount()).isEqualTo(1);
+    Link link = protoSpan.getLinks(0);
+    assertThat(link.getTraceId()).isNotEmpty();
+    assertThat(link.getSpanId()).isNotEmpty();
```

Comment:
If we save off the generated ids, we could easily assert them here, right?

Reply:
Yup!